### ISSUE TITLE
Values greater than 2^31 cause integer overflow error in LK callbacks

### ISSUE THREAD

**Issue by brtietz** [NREL/SAM#496]
Putting 1,000,000,000 in strings in parallel in a PV subarray or 1.7e7 in battery capacity will cause the resulting calculated fields in LK to overflow into negative numbers.  Update paths in SAM and or LK repositories to ensure all relevant functions use size_t instead of int.

Occurs in 2020.11.29, and possibly additional prior versions.

**Comment by sjanzou** [comment]
Specific use case for initial resolution PB-Batt/Residential - battery capacity 2.4e7 is fine:
![image](https://user-images.githubusercontent.com/6498311/105837986-b44df480-5f8c-11eb-9ba3-5abc3ef482dc.png)

battery capacity 2.5e7 causes overflow of integer to negative number of cells 
![image](https://user-images.githubusercontent.com/6498311/105838098-e1020c00-5f8c-11eb-949e-67e0455427a1.png)

**Comment by sjanzou** [comment]
Initial number of cells displayed on battery page addressed (release on left and SAM_496 branch build on right):
![image](https://user-images.githubusercontent.com/6498311/105842459-71434f80-5f93-11eb-95f4-782065b95398.png)

**Comment by sjanzou** [comment]
@cpaulgilman, @brtietz - While going through this issue, I discovered the data tables output for PV-Batt / Res configuration includes Fuel cell outputs 
![Uploading image.png…]()

**Comment by brtietz** [comment]
Good catch! I filed a new issue for the fuel cell outputs here: https://github.com/NREL/SAM/issues/510

**Comment by brtietz** [comment]
SAM file showing the original issue: [SAMBatteryPowerTargetProblem.zip](https://github.com/NREL/SAM/files/5875045/SAMBatteryPowerTargetProblem.zip)

**Comment by sjanzou** [comment]
Fixed with pull request 541 in ssc and pull request 543 in SAM with no changes in default results.
Additional test added for ssc in cmod_overflow_SAM496_test.cpp and input_cases/json/SSC_Test_Case_SAM496.json from .sam project file 
[SSC Test Case SAM496.zip](https://github.com/NREL/SAM/files/6022009/SSC.Test.Case.SAM496.zip)
More details in the 34-page Word document
[GitHubIssuesSAM_496.docx](https://github.com/NREL/SAM/files/6022082/GitHubIssuesSAM_496.docx)

**Comment by brtietz** [comment]
As of 12/13/23, this can still be tripped with 240,000,000,000.000 kWh batteries.

**Comment by cpaulgilman** [comment]
This seems to be resolved in SAM 2023.12.17. @brtietz OK to close this?

**Comment by sjanzou** [comment]
> This seems to be resolved in SAM 2023.12.17. @brtietz OK to close this?
> 
> ![image](https://private-user-images.githubusercontent.com/27711933/306816235-59b959cd-446b-4c9e-8014-8d683e59c97c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDg1ODg1MjIsIm5iZiI6MTcwODU4ODIyMiwicGF0aCI6Ii8yNzcxMTkzMy8zMDY4MTYyMzUtNTliOTU5Y2QtNDQ2Yi00YzllLTgwMTQtOGQ2ODNlNTljOTdjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjIyVDA3NTAyMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTEyMDhhMmIwZGQwYzE0ZTdlMTE4MzAwODg0MmE2MTk5ZTA0YjhmYjMzN2I2NWI3YjU0MGNjM2VkYWVjMTNiYzkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.OrAPLT4lrNlQ0O7__Zp47IIOaNlooZJfzEn-tGiJ_pI)

Working for me in the latest patch branch builds
![image](https://github.com/NREL/SAM/assets/6498311/95c3d98c-4a1f-49a5-b11c-b03a29b46624)

**Comment by sjanzou** [comment]
4/16/2024 Working with 240,000,000,000,000.000 in latest patch branches
![image](https://github.com/NREL/SAM/assets/6498311/1cea70a9-49cd-49fb-a021-ef028f3b9c79)

Closing issue.

@brtietz, please reopen if you run into any issues!
